dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -133,10 +146,212 @@ xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[VIP] ad
 - In a non-cloud environment, by configuring ingressIP range (`ingressIPNetworkCIDR`),
 `service.type=LoadBalancer`, and `service.port.ingressIP`.
 
-The administrator must ensure the external IPs are routed to the nodes and local firewall rules on all nodes allow access to the open port.
+The administrator must ensure the external IPs are routed to the nodes and local
+firewall rules on all nodes allow access to the open port.
+
+.Example External IP Configured within a Service Definition
+----
+$ oc create -f - <<INGRESS
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgresql-ingress
+spec:
+  externalIPs:
+  - 10.9.54.100 <1>
+  ports:
+  - port: 5432
+    protocol: TCP
+  selector:
+    name: postgresql
+  type: LoadBalancer
+INGRESS
+----
+<1> The IP address assigned to the external IP.
+
+The external IP address is not managed by the underlying Kubernetes
+infrastructure and must be maintained and provided by a cluster administrator.
+While external IPs provide a solution for accessing services on the {product-tlte}
+cluster, there are several shortcomings:
+
+* The cluster administrator user must ensure the external IP is not in any range
+that the cluster can use. The user needs to work with the administrator that
+controls the network external to the cluster to assign the external IP address,
+a DNS name for the IP, and ensure traffic to the IP can get to the nodes in the
+cluster. The cluster administrator must ensure firewall rules permit the packets
+to get into the nodes.
+* There are no protections in place to restrict the usage of a particular external
+address configured within the cluster. This allows the potential for a single
+address to be used by multiple services targeting the same port. When this
+situation occurs, the service which requested the port first is given use of the
+port.
+
+xref:using-ingress-IP-self-service[Ingress IP Self-Service] provides a solution.
+
+[[using-ingress-IP-self-service]]
+=== Using Ingress IP Self-Service
+
+Ingress IP Self-Service streamlines the allocation of External IPs for accessing
+services in the cluster. Cluster administrators can designate a range of
+addresses using a CIDR notation, which allows an application user to make a
+request against the cluster for an external IP address. When a service is
+configured with the type `LoadBalancer`, an External IP address will be
+automatically assigned from the designated range.
+
+[NOTE]
+====
+Ingress IP Self-Service is only applicable in non-cloud based environments.
+====
+
```

Comment:
We only set up the rules to allow the traffic to get in if it arrives on the external IP and port.  If they have other rules that block access to the port, then they need to relax those.

The point is to allow people to self-provision external IP addresses.  Normally that requires cluster admin privs, but by turning it into LoadBalancer services, users can create them (subject to quota restrictions).

As to the DNS names, yes... the user would have to examine the service and update DNS.  Ideally they would have something that they ran that watched the service and saw the externalIP and then updated DNS whenever there was a change (and that would be a nice sample container and docs to provide).  But initially people will probably do it manually.